src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,10 +31,34 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+       value = address;
+        initializeAddressParts(address);
     }
 
     /**
+     * Initialises Block, Street, Unit, PostalCode members of Address class
+     * by splitting and assigning given Address string
+     * according to regex
+     */
+    private void initializeAddressParts(String address){
+    	String[] addressParts = address.split(ADDRESS_SPLIT_REGEX); //splits address string according to regex
+    	
+    	//assigns address parts according to number of parts
+    	switch(addressParts.length){
```

Comment:
Indentation problem here?. Our coding standard requires you to use 4 spaces instead of tabs. You can configure Eclipse to convert tabs to spaces automatically.